### ISSUE TITLE
Move temporal classes to "base" mini-library

### ISSUE THREAD

**Issue by navispatial** [navibyte/geospatial#69]
Move `Instant` and `Interval` classes from `lib/src/common/temporal` to `lib/src/base/temporal`.

After this `lib/src/base` contains:
* geometry
* temporal

Description of the mini-library  `package:geocore/base.dart`:
```dart
/// Base classes for geospatial geometries and temporal objects.
/// 
/// *Geometry* classes include coordinates, points, bounds, point series, and 
/// transform and projection abstractions.
/// 
/// *Temporal* classes include temporal instants and intervals.
///
/// Usage: import `package:geocore/base.dart`
library base;

export 'src/base/geometry.dart';
export 'src/base/temporal.dart';
```

Documentation of mini-libraries after the change:

Library         | Export also | Description 
--------------- | ----------- | -----------------------------------------------
**base**        | | Base classes for geospatial geometries and temporal objects.
**coordinates** | base | Cartesian and geographic points with some common coordinate transforms.
**data**        | base, coordinates | Geospatial features and geometries (linestring, polygon, multi geometries).
**parse**       | base, coordinates, data | GeoJSON and WKT (Well-known text representation of geometry) data parsers.
**proj4d**      | base | Projections provided by the external [proj4dart](https://pub.dev/packages/proj4dart) package.

**Comment by navispatial** [comment]
Relates also to : [Define consistent mini library exports with base classes included #31](https://github.com/navibyte/geospatial/issues/31)

**Comment by navispatial** [comment]
Also renamed `base/geometry` to `base/spatial`

**Comment by navispatial** [comment]
Implemented in pre-release `0.8.0-a.7` of the upcoming BETA-version `0.8.0`
